qadence2_expressions/core/expression.py

```diff
@@ -489,6 +490,18 @@ def __pow__(self, other: object) -> Expression:
         if other.is_one:
             return self
 
+        # Power of power is a simple operation and can be evaluated here.
+        if (
+            self.is_quantum_operator
```

Comment:
Hm there is a number this checks sometimes. Just wondering whether they couldn't be grouped under fewer (possibly single) predicate(s).

Reply:
can you point where you found them out? Probably we could group them into a single function